mypy/fastparse2.py

```diff
@@ -819,11 +820,16 @@ def visit_Str(self, s: ast27.Str) -> Node:
             contents = str(n)[2:-1]
 
             if self.pyversion[0] >= 3:
+                assert False  # Shouldn't get here with an ast27.Str
                 return BytesExpr(contents)
             else:
-                return StrExpr(contents)
+                if s.has_b:
```

Comment:
I believe(*) this block is the full extent of the changes needed in mypy to differentiate between `b''` and `''` -- all the heavy lifting is done by dropbox/typed_ast#17 (thanks @ilevkivskyi !).

And if typeshed defines bytes as an alias for str the treatment of StrExpr and BytesExpr is identical -- however the intent is for typeshed to make bytes and str separate classes, and then it's important that the inferred type of `b''` is bytes while that of `''` is str.

(*) Well there may be a few more places where a hardcoded check for builtins.str is used that will have to also allow builtins.bytes, but in many places builtins.str is actually still the only type to be treated so. E.g. `dict(x=1)` always has str keys, never bytes keys.